packages/api-plugin-fulfillment-method-shipping-ups/src/getFulfillmentMethodsWithQuotesShippingUPS.js

```diff
@@ -0,0 +1,77 @@
+import calculateUPSRate from "./util/calculateUPSRate.js";
+
+const packageName = "fulfillment-method-shipping-ups";
+const fulfillmentTypeName = "shipping";
+const fulfillmentMethodName = "ups";
+
+/**
+ * @summary Returns a list of fulfillment method quotes based on the items in a fulfillment group.
+ * @param {Object} context - Context
+ * @param {Object} commonOrder - details about the purchase a user wants to make.
+ * @param {Array} [previousQueryResults] - an array of shipping rates and
+ * info about failed calls to the APIs of some shipping methods providers
+ * e.g Shippo.
+ * @returns {Array} - an array that contains two arrays: the first array will
+ * be an updated list of shipping rates, and the second will contain info for
+ * retrying this specific package if any errors occurred while retrieving the
+ * shipping rates.
+ * @private
+ */
+export default async function getFulfillmentMethodsWithQuotesShippingUPS(context, commonOrder, previousQueryResults = []) {
+  const { collections } = context;
+  const { Fulfillment } = collections;
+  const [rates = [], retrialTargets = []] = previousQueryResults;
+  const currentMethodInfo = { packageName };
+
+  if (retrialTargets.length > 0) {
+    const isNotAmongFailedRequests = retrialTargets.every((target) => target.packageName !== packageName);
+    if (isNotAmongFailedRequests) {
+      return previousQueryResults;
+    }
+  }
+
+  // TODO This has to be a flag in Fulfillment collection
```

Comment:
this much be copy pasta? Is this relevant to this code at all?

Reply:
Removed the commented code. Added validation for shipping-UPS which is based on the entry `"provider.enabled": true` in Fulfillment collection instead of checking appSettings.

Reply:
Updated the other two ff-method functions also
```
getFulfillmentMethodsWithQuotesShippingFlatRate
getFulfillmentMethodsWithQuotesPickupStore
```